src/schema.ts

```diff
@@ -147,79 +89,79 @@ export interface VibraphoneDropEvent {
 	 */
 	channel: VibraphoneChannel;
 }
-export enum EventBakeType {
-	AUTO,
-	MODIFIED_AUTO,
-	MANUAL,
-}
-export type PerformanceDropEvent = DropEvent & CorePerformanceDropEvent;
-export interface CorePerformanceDropEvent {
-	bakeType: EventBakeType;
+
+/** Represents the dropping of a single marble with no timing information. */
+export type DropEvent = BassDropEvent | DrumDropEvent | VibraphoneDropEvent;
+
+/** Represents information common to all drop events. */
+export interface CoreDropEvent {
+	/** The tick (pulse) that the marble is to be dropped on. */
+	tick: number;
 }
 
-/**
- * Represents a single performance of the associated [[Program]].
- *
- * Performances contain all events that exist outside of the
- * programming wheel. This could be a manual drop, a change of
- * one of the bass capos, or any other event in the [[Event]] type.
- */
-export interface Performance {
-	/** Information about this performance. */
-	metadata: PerformanceMetadata;
-	/** The program associated with this performance. */
-	program: Program;
+/** Represents the dropping of a single marble with an associated tick. */
+export type TickedDropEvent = CoreDropEvent & DropEvent;
+
+//#endregion
+
+//#region Program
+
+/** Represents metadata about a program. Used to indicate how this program should be played. */
+export interface ProgramMetadata {
 	/**
-	 * The state of the machine at the start of the performance.
-	 *
-	 * Note that this should not be changed during the performance.
-	 * Use [[Program.state]] if you want to keep track of the current
-	 * state of the machine. This means that [[Performance.initialState]]
-	 * should be copied into [[Program.state]] at tick 0 of Performance playback.
+	 * The title of this program. It can be distinct from the title of the performance.
 	 */
-	readonly initialState: State;
```

Comment:
Totally unnecessary IMO, especially since you're stating that it is the program state at tick 0.

Reply:
Not totally unnecessary, the workflow that has been outlined by a few outspoken people here means a state is included in a Program, removing this would require additional processing when creating a performance.  Easy to do though I admit

Reply:
Got it. I didn't know. Thanks.